### ISSUE TITLE
Feature importance TAN

### ISSUE THREAD

**Issue by PARODBE** [pgmpy/pgmpy#1490]
Hi!,

I am using a TAN model in order to resolver a problem. Do you have any idea for calculate feature importance of TAN model? I have tryied to use SHAP with pgmpy but it hasn't worked for me.

Thanks!

**Comment by ankurankan** [comment]
@PARODBE Sorry for the late reply, I was on holiday. I think it would depend on your application. A couple of ways I can think are:
1. If we limit ourselves to the Markov blanket of the variable we are trying to predict, calculating the cross-entropy between its conditional and marginal distribution should give us the amount of information that specific variables add. Essentially, taking the conditional distribution of the dependent variable, removing conditioning variables, and computing the cross-entropy. This can then be interpreted as variables adding more information as more important.
2. Another option could be to use the causal inference framework to compute the direct or total effect (depending on the application) of different variables on the variable we are trying to predict and use higher effect variables as more important.

**Comment by ankurankan** [comment]
Also found this: https://hal-brgm.archives-ouvertes.fr/hal-02408006/document which is from a sensitivity analysis approach (if I remember correctly SHAP also uses the same approach) which you might find interesting.

**Comment by PARODBE** [comment]
uooh Thank you, although It would be some of code for understand better, because in python I haven't found any explanaition of this..but thank you for the information.

**Comment by ankurankan** [comment]
@PARODBE Yeah, code examples would be difficult to find for this. I will try to write some basic examples on these in the coming few days and will send it to you.

**Comment by PARODBE** [comment]
Uoooo thank you! @ankurankan it would be great!
